app/latest_image/page.jsx

```diff
@@ -0,0 +1,83 @@
+"use client";
+import { useEffect, useState } from "react";
+import styles from "./page.module.css";
+
+export default function EpicImage() {
+  const [image, setImage] = useState(null);
+  const [error, setError] = useState(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    async function fetchLatestImage() {
+      setLoading(true);
+      setError(null);
+      try {
+        const url = `https://api.nasa.gov/EPIC/api/natural/images?api_key=${process.env.NEXT_PUBLIC_API_KEY}`;
+        const response = await fetch(url);
+        const data = await response.json();
+
+        if (data && data.length > 0) {
+          const latestImage = data[0];
+          const latestDate = latestImage.date;
+
+          const detailsUrl = `https://api.nasa.gov/EPIC/api/natural/date/${latestDate}?api_key=${process.env.NEXT_PUBLIC_API_KEY}`;
+          const detailsResponse = await fetch(detailsUrl);
+          const detailsData = await detailsResponse.json();
+
+          if (detailsData && detailsData.length > 0) {
+            setImage(detailsData[0]);
+          } else {
+            setImage(null);
+          }
+        } else {
+          setImage(null);
+        }
+      } catch (error) {
+        setError(error.message);
+      } finally {
+        setLoading(false);
+      }
+    }
+
+    fetchLatestImage();
+  }, []);
+
+  if (error) {
+    return <div>Error: {error}</div>;
+  }
+  if (loading) {
```

Comment:
I'm curious, is it better to replace the whole div like this or just the `<picture>...</picture>` with the Loading, Error or No Image component?